packages/vsx-registry/src/common/vsx-registry-api.ts

```diff
@@ -68,13 +69,34 @@ export class VSXRegistryAPI {
         return this.fetchJson(apiUri.resolve(id.replace('.', '/')).toString());
     }
 
+    /**
+     * Get the latest compatible extension version.
+     * - an extension satisfies compatibility if its `engines.vscode` version is supported.
+     * @param id the extension id.
+     *
+     * @returns the data for the latest compatible extension version if available, else `undefined`.
+     */
+    async getLatestCompatibleExtension(id: string): Promise<VSXExtensionRaw | undefined> {
+        const extension = await this.getExtension(id);
+        for (const extensionVersion in extension.allVersions) {
```

Comment:
I wonder should not it be rather solved in the registry? For each extension we will do potentially many server requests. I imagine there should be one request to fetch an extension by id for such vscode engine.

Reply:
> I wonder should not it be rather solved in the registry? For each extension we will do potentially many server requests. I imagine there should be one request to fetch an extension by id for such vscode engine.

It would be good to enhance the registry API to also accept the `engines.vscode` as an optional query parameter and only return the latest compatible result. 

Would you be fine to improve the API to accept such a query parameter and add the logic to determine the latest compatible extension directly in the registry?

Reply:
cc @spoenemann

Reply:
I wonder if e.g. VScodium, that now uses Open VSX though an adapter, is able to pick a version of an extension to install, that is appropriate to it's own API version?  Presumably this would be done client-side too? Potentially in a better optimised way?